### DISCUSSION TITLE
Question: Safety of FP with Java?

### DISCUSSION THREAD

**Discussion by autoca** [freeplane/freeplane#659] [Help] [answered]
Hi again,
I now noticed that the setup file of Freeplane with Java included does not seem to use the latest Java version. Since other people say that one should always use the latest JRE version, I am wondering, can other programs use that JRE that is installed with this Freeplane setup or is this encapsulated so that it can only be used by Freeplane itself?

(I tried to install the latest JRE and use the Freeplane setup without Java, setting the [EDIT: FREEPLANE_JAVA_HOME] variable and rebooting and all, but it still gives an error message when trying to run FP.)

Regards, a.

**Comment by dpolivaev** [comment]
Freeplane uses Java 15 instead of Java 17 because of the reasons discussed in https://github.com/freeplane/freeplane/discussions/275. I take Java from https://www.azul.com/downloads/?version=java-15-mts&os=windows&package=jre which still gets updates, but the embedded Java runtime is not automatically updated after you install it. 

If you want to use an external Java version you need to set the environment variable `FREEPLANE_JAVA_HOME` as described in https://docs.freeplane.org/getting-started/Command-line_options_and_configuration.html#select-java-installation-used-to-run-freeplane.

Having a docu or not having a docu makes almost no difference: it is hard to find the answer until you already know what you are looking for.

Cheers

**Reply by autoca** [reply]
Thank you for the side infos.
My question, though, mainly was: Can other programs access the "embedded" Java version of Freeplane or does "embedded" really mean that only Freeplane can use it?

Regarding the environment variable, yes, sorry, I wrongly wrote PATH up above but I really meant the environment variable FREEPLANE_JAVA_HOME as in the description you linked to. I set it like that but it did not work. But depending on the answer to the question above, it may not matter.

**Reply by dpolivaev** [reply]
If you ask more than one question in a single post, I am highly likely to oversee any number of them.
In order to use the embedded Java version, other programs need to know where to find it. I do not know how they can be so smart.

**Reply by autoca** [reply]
Ok, I see, thank you.

**Comment by autoca** [comment, marked as answer]
Ok, I found the solution. The latet JRE updates from Oracle cannot be used for free anymore. I assume that that is the problem, because I just installed the latest version from Oracle without reading these side notes.
Furtunately, Wikipedia says: "Due to Oracle no longer releasing updates for long-term support (LTS) releases under a permissive license, others have begun offering builds. Linux distributions have always offered their own builds, and Windows also offers one now;"
https://en.wikipedia.org/wiki/OpenJDK#OpenJDK_builds

The windows build of OpenJDK can be found here:
https://docs.microsoft.com/en-us/java/openjdk/install

I installed it, set the FREEPLANE_JAVA_HOME env variable to the correct path and, voilà, Freeplane is humming like a well oiled sports car with a brand new Java 17 version under the hood. (Btw, no Windows restart needed.)

Regards, a.

**Reply by macmarrum** [reply]
Just to make people aware, there are many providers of OpenJDK binaries for Windows. As you have found, Microsoft is one of them. Another is Azul. _BTW, Freeplane bundles binaries from Azul in Freeplane-Setup-with-Java-1.10.x.exe, as mentioned by Dimitry_

- https://www.azul.com/downloads/?os=windows&architecture=x86-64-bit&package=jre#download-openjdk

Adoptium is also a popular choice:

- https://adoptium.net/temurin/releases
